### PR TITLE
Read-only JSON API

### PR DESCRIPTION
**Pull request by hunterboerner**
This pull request documents the implementation of a [JSON API 1.0](jsonapi.org) compliant API.

Major changes:

~~Remodel was swapped out for Relax. This is an Elixir package for JSON API serialization and will be updated to the 1.0 spec soon.~~ Relax was swapped out for ja_serializer which is the serializer extracted from Relax and updated to JSON API 1.0.

### CODE REVIEW

File: mix.lock [hunk 1]
```diff
@@ -2,24 +2,28 @@
...
-  "excoveralls": {:hex, :excoveralls, "0.3.9"},
+  "excoveralls": {:hex, :excoveralls, "0.3.10"},
   "exjsx": {:hex, :exjsx, "3.1.0"},
-  "exq": {:git, "git://github.com/akira/exq.git", "2c1cdc731611e3ee1689c984333edc9d114952a3", []},
+  "exq": {:git, "git://github.com/akira/exq.git", "4fe65d7de524fc7720ee1962fa4de75c33fc7210", []},
   "hackney": {:hex, :hackney, "1.1.0"},
   "idna": {:hex, :idna, "1.0.2"},
+  "inflex": {:hex, :inflex, "0.2.9"},
+  "ja_serializer": {:git, "git://github.com/AgilionApps/ja_serializers.git", "372d3e0d036c2fe04e0b7ef678d1d292299f22c7", []},
+  "ja_serializers": {:git, "git://github.com/AgilionApps/ja_serializers.git", "3bdbb70c670e2d35a75748e0b04d7b3293548407", []},
```
Comment (alanpeabody) [inline mix.lock]: Looks like when I renamed the ja_serializer repo to a singular name I caused this.

**Comment by hunterboerner** [comment]
All looks well on my end. @alanpeabody do my serializers look alright? Have you implemented any phoenix integrations?

**Comment by alanpeabody** [comment]
@hunterboerner Do you mean to include both the link and the id/type for every relationship? I usually only do one or the other.

File: apps/rabbitci_core/web/serializers/branch_serializer.ex [hunk 2]
```diff
@@ -0,0 +1,26 @@
+defmodule RabbitCICore.BranchSerializer do
+  use JaSerializer
+
+  require RabbitCICore.SerializerHelpers
+  alias RabbitCICore.SerializerHelpers
+
+  serialize "branches" do
+    attributes [:updated_at, :inserted_at, :name]
+    has_one :project, link: ":project_link", field: :project_id, type: "projects"
+    has_many :builds, include: RabbitCICore.BuildSerializer, link: ":builds_link"
```
Comment (alanpeabody) [inline apps/rabbitci_core/web/serializers/branch_serializer.ex]: The link here should be fine as an atom `link: :project_link`.

Would being able to set the link via an anonymous function be a nice improvement? (it is not possible now)

``` elixir
import RabbitCICore.Router.Helpers

  has_one :project, 
    link: &(project_path(&2, :show, &1.project_id)),
    type: "project",
    field: :project_id
```

**Comment by alanpeabody** [comment]
@hunterboerner I have not done any Phoenix integration yet, but if you have specific requests I will see what I can do.

**Comment by hunterboerner** [comment]
> @hunterboerner Do you mean to include both the link and the id/type for every relationship? I usually only do one or the other.

I think when I flatten the API (nested APIs are a PITA with ember) I'll remove a lot of the links.

> @hunterboerner I have not done any Phoenix integration yet, but if you have specific requests I will see what I can do.

Link helpers I guess. It would also be nice to have bidirectional serialization.

**Comment by hunterboerner** [comment]
Everything looks alright to me so I'm going to merge this.

**Pull request merged**